client/sites-dashboard/components/sites-ellipsis-menu.tsx

```diff
@@ -243,8 +244,44 @@ const SiteDropdownMenu = styled( DropdownMenu )( {
 function useSubmenuItems( site: SiteExcerptData ) {
 	const { __ } = useI18n();
 	const siteSlug = site.slug;
+	const stagingSite = site.is_wpcom_staging_site;
+	const isStagingSiteEnabled = isEnabled( 'yolo/staging-sites-i1' );
 
 	return useMemo< { label: string; href: string; sectionName: string }[] >( () => {
+		if ( stagingSite || ! isStagingSiteEnabled ) {
+			return [
+				{
+					label: __( 'SFTP/SSH credentials' ),
+					href: `/hosting-config/${ siteSlug }#sftp-credentials`,
+					sectionName: 'sftp_credentials',
+				},
+				{
+					label: __( 'Database access' ),
+					href: `/hosting-config/${ siteSlug }#database-access`,
+					sectionName: 'database_access',
+				},
+				{
+					label: __( 'Deploy from GitHub' ),
+					href: `/hosting-config/${ siteSlug }#connect-github`,
+					sectionName: 'connect_github',
+				},
+				{
+					label: __( 'Web server settings' ),
+					href: `/hosting-config/${ siteSlug }#web-server-settings`,
+					sectionName: 'web_server_settings',
+				},
+				{
+					label: __( 'Clear cache' ),
+					href: `/hosting-config/${ siteSlug }#cache`,
+					sectionName: 'cache',
+				},
+				{
+					label: __( 'Web server logs' ),
+					href: `/hosting-config/${ siteSlug }#web-server-logs`,
+					sectionName: 'logs',
+				},
+			];
+		}
 		return [
 			{
 				label: __( 'SFTP/SSH credentials' ),
```

Comment:
This works well. The only downside is that we are defining almost the same menu twice. I want to suggest an optimization. 

When I had to deal with multiple different conditions, I created this pattern with conditions and a filter.
Could you try to refactor your code and re-use it?

https://github.com/Automattic/wp-calypso/pull/73356/files#diff-173bec7d847197dc6a90c47661016102ba95728dab618e17b774c11c7ed1994cR258-R310

Reply:
It took me a bit of tinkering around but I think it looks better now - let me know what you think!